environments/common/inventory/group_vars/all/nfs.yml

```diff
@@ -3,7 +3,7 @@
 # See: https://github.com/stackhpc/ansible-role-cluster-nfs
 # for variable definitions
 
-nfs_server_default: "{{ hostvars[groups['control'] | first ].internal_address }}"
```

Comment:
Should we default internal address to ansible_hostname? I had intended that to be set to either an ip address or hostname to control this kind of behaviour i.e it is the address used for internal communication beween servers e.g serverb connects to servera.internal_address, so by setting internal address we'd always use the hostname by default, but some could customise it use ansible_host if they didn't have working DNS.

Reply:
Might need an actual discussion to make sure I get the implications - in TF we also set some network vars, need to make sure we've got a consistent/docu'd approach here.

Reply:
So:
- Don't want to use `inventory_hostname` as this might not be routable w/o DNS
- Don't want to use `ansible_host` as this might be e.g. on management network not suitable for storage etc.
- TODO: check if `ansible_host` defaults to `inventory_hostname` if not defined
- Currently, TF tends to define:
 
      server_networks='${jsonencode({for net in control.network: net.name => [ net.fixed_ip_v4 ] })}'

   (based on p3-appliances). This is nice and general-purpose, e.g. for defining multihomed Lustre servers etc, but seems a bit complex.
- Considering defining `internal_address` defaulted to `inventory_hostname` in inventory. This could be overriden in environment's `hosts` file (e.g. with `ansible_host` or another network) if necessary.
- Needs documentation!

@jovial that summary ok? I will push an `internal_address` approach when I can.

Reply:
Looks good to me. Agree that setting internal_address seems like the cleanest solution for now.

Reply:
> check if ansible_host defaults to inventory_hostname if not defined

Yes, it does.

Reply:
@jovial we also have `api_address` in `environments/common/inventory/group_vars/all/defaults.yml` which is currently defaulted to ansible_host - should we change this too?